distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Date.enso

```diff
@@ -431,9 +438,17 @@ type Date
              from Standard.Base import Date
              import Standard.Base.Data.Time.Duration
 
-             example_subtract = Date.new 2020 - 7.days
-    - : Date_Period -> Date
-    - self _ = unimplemented "Date.-"
+             example_subtract = Date.new 2020 - (Period.new days=7)
+    - : Period -> Date ! (Time_Error | Illegal_Argument_Error)
+    - self period =
+        case period of
+            _ : Period.Period ->
+                new_java_date = Time_Utils.date_adjust self Time_Utils.AdjustOp.MINUS period.internal_period
+                new new_java_date.year new_java_date.month new_java_date.day
+            _ : Duration.Duration ->
+                Error.throw (Time_Error_Data "Date does not support time intervals")
```

Comment:
I'd suggest adding the word `Duration` somewhere to this error message, so that it's clearer that the type is wrong. Period is a time interval too to some interpretations, so the current message can be confusing. Maybe just `Date doe snot support adding/subtracting Duration. Use Period instead.`?